Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -251,6 +254,139 @@ public AndConstraint<ObjectAssertions> NotBe(object unexpected, string because =
             return new AndConstraint<ObjectAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that an object is enumerable using a foreach loop.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase explaining why the assertion should be satisfied. If the phrase does not
+        /// start with the word <i>because</i>, it is prepended to the message.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more values to use for filling in any <see cref="string.Format(string,object[])" /> compatible placeholders.
+        /// </param>
+        public AndWhichConstraint<ObjectAssertions, IEnumerable> BeEnumerable(string because = "", params object[] becauseArgs)
+        {
+            AssertIsEnumerable(Subject, because, becauseArgs);
+
+            return new AndWhichConstraint<ObjectAssertions, IEnumerable>(this, new ObjectEnumerable(Subject));
+        }
+
+        public AndConstraint<ObjectAssertions> BeEnumerable(IEnumerable expected, Func<object, object, bool> equalityComparison, string because = "", params object[] becauseArgs)
+        {
+            Guard.ThrowIfArgumentIsNull(equalityComparison, nameof(equalityComparison));
+
+            bool subjectIsNull = Subject is null;
+            bool expectationIsNull = expected is null;
+            if (!(subjectIsNull && expectationIsNull))
+            {
+                Guard.ThrowIfArgumentIsNull(expected, nameof(expected), "Cannot compare collection with <null>.");
+
+                if (subjectIsNull)
+                {
+                    Execute.Assertion
+                        .BecauseOf(because, becauseArgs)
+                        .FailWith("Expected {context:collection} to be equal to {0}{reason}, but found <null>.", expected);
+                }
+
+                AssertIsEnumerable(Subject, because, becauseArgs);
+
+                var subjectType = Subject.GetType();
+
+                // assert equality using the public GetEnumerator()
+                AssertEquality(Subject, subjectType, expected, equalityComparison, because, becauseArgs);
+
+                // assert equality using the GetEnumerator() of implemented enumerable interfaces
```

Comment:
I wasn't expecting this method to enumerate through all supported interfaces, but I do realize the confusion when an object has multiple ways of enumerating. Not sure yet how to deal with this. cc @jnyrup